modules/misunderstood/src/backend/misunderstood-unit.test.ts

```diff
@@ -0,0 +1,115 @@
+import * as sdk from 'botpress/sdk'
+
+import 'jest'
+import { addQnA, addNLU, removeNLU } from './applyChanges'
+import { FLAG_REASON } from '../types'
+
+const makeMockGhost = (fileData: any = {}, directories: string[] = []) => {
+  return {
+    upsertFile: jest.fn(
+      (rootFolder: string, file: string, content: string | Buffer, options?: sdk.UpsertOptions): Promise<void> => {
+        return
+      }
+    ),
+    readFileAsObject: jest.fn(
+      async <T>(rootFolder: string, file: string): Promise<T> => {
+        // Do a deep copy
+        return JSON.parse(JSON.stringify(fileData)) as T
+      }
+    ),
+    directoryListing: jest.fn(
+      async (
+        rootFolder: string,
+        fileEndingPattern: string,
+        exclude?: string | string[],
+        includeDotFiles?: boolean,
+        options?: sdk.DirectoryListingOptions
+      ): Promise<string[]> => {
+        return directories
+      }
+    )
+  }
+}
+
+describe('applyChanges', () => {
+  it('runs addQnA', async () => {
```

Comment:
the names `runs addQnA/addNLU` feels weird, sounds like you aren't testing the full behaviour of the function. For unit tests on function I usually call it the function name directly, for integration tests I write a full description (like you do above). Let me know what you think!

Reply:
That works for me, I'll make the change.